django/shortcuts.py

```diff
@@ -62,7 +62,7 @@ def _get_queryset(klass):
     return klass
 
 
-def get_object_or_404(klass, *args, **kwargs):
+def get_object_or_404(klass, exception_detail: str = None, *args, **kwargs):
```

Comment:
Django does not use type annotations, so I don't think we'd want this here.

Reply:
@adamzap why do you think that changing signature is pretty significant since is an optional value? maybe it could be necessary update documentation but I think that it could be useful. Many times i wrote custom exceptions to specify custom detail in getting object, this feature could avoid this!

Reply:
That's a fair question. I think it's more because this wouldn't be considered a trivial change in Django, so it would require an accepted [Trac](https://code.djangoproject.com/query) ticket.

I'm curious, are you trying to enhance the message on the 404 page that shows when `DEBUG` is set to `True`?